### PR TITLE
WDL 2.0: min and max functions [BA-6158]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Implements WDL spec https://github.com/openwdl/wdl/pull/304.

### CODE REVIEW

File: centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl [hunk 1]
```diff
@@ -15,8 +15,15 @@ workflow biscayne_as_map_et_al {
 
   Array[Pair[String,Int]] z_pairs_in = [("a", 1), ("b", 2), ("a", 3)]
 
+  Int smallestInt = 1
+  Float smallFloat = 2.718
+  Float bigFloat = 3.141
+  Int biggestInt = 10
```
Comment (aednichols) [inline centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl]: Does it make sense to test some extremes like int max and float max? Which is larger, int max or float max? I don't know the answer, but we should do the right thing.

File: centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl [hunk 2]
```diff
@@ -28,6 +35,14 @@ workflow biscayne_as_map_et_al {
 
     Map[String,Array[Int]] x_collect_by_key_out = collect_by_key(x_pairs_in) # {"a": [1], "b": [2], "c": [3]}
     Map[String,Array[Int]] z_collect_by_key_out = collect_by_key(z_pairs_in) # {"a": [1, 3], "b": [2]}
+
+    # min(), max():
+    # =================================================
```
Comment (aednichols) [inline centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl]: These section headers are really helpful 👍

File: wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala [hunk 3]
```diff
@@ -106,4 +106,49 @@ object BiscayneValueEvaluators {
...
+      (value1, value2) flatMapN {
+        case (v1, v2) =>
+          val newValue = (v1.value, v2.value) match {
+            case (WomInteger(i1), WomInteger(i2)) => WomInteger(Math.max(i1, i2)).validNel
+            case (WomInteger(i1), WomFloat(l2)) => WomFloat(Math.max(i1.doubleValue, l2)).validNel
+            case (WomFloat(l1), WomInteger(i2)) => WomFloat(Math.max(l1, i2.doubleValue)).validNel
+            case (WomFloat(l1), WomFloat(l2)) => WomFloat(Math.max(l1, l2)).validNel
+            case (other1, other2) => s"Invalid arguments to 'max':(${other1.typeName}, ${other2.typeName})".invalidNel
+          }
+          newValue map { v => EvaluatedValue(v, v1.sideEffectFiles ++ v2.sideEffectFiles) }
```
Comment (mcovarr) [inline wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala]: another try to DRY

File: centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl (hunk 1 above)
Comment (cjllanwarne) [reply centaur/src/main/resources/standardTestCases/wdl_biscayne/biscayne_new_engine_functions/biscayne_new_engine_functions.wdl]: I added the test. It turns out that max float is significantly larger than max int, and the max-ing of int X float gives you float, so I think we're safe.

File: wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala (hunk 3 above)
Comment (cjllanwarne) [reply wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala]: NO TIME TO DRY

**Pull request merged**